### PR TITLE
Load Adapt unconditionally for Julia 1.8

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

**Comment by vchuravy** [comment]
Needs backport to EnzymeCore 0.6

**Comment by wsmoses** [comment]
Some folks in the past were hesitant about adding EnzymeCore as a dep due to Adapt as a sub-dependency, so I'm a bit hesitant here.

**Comment by vchuravy** [comment]
So the question is what is their concern? Is it load time? If so package extensions solves that but it can only solve it on 1.9+

I strongly think that we shouldn't have broken functionality on 1.8 and lower

File: lib/EnzymeCore/src/EnzymeCore.jl [hunk 1]
```diff
@@ -257,4 +257,9 @@ function compiler_job_from_backend end
 
 include("rules.jl")
 
+if !isdefined(Base, :get_extension)
```
Comment (vchuravy) [inline lib/EnzymeCore/src/EnzymeCore.jl:260]: @wsmoses the original PR by @devmotion had the backwards compatibility https://github.com/EnzymeAD/Enzyme.jl/pull/954/files#diff-94319485f88a6e9c6ce4669ce81ce9d3a46dab48d8c9e579b22272fc798a39a9R172

File: lib/EnzymeCore/Project.toml [hunk 2]
```diff
@@ -7,11 +7,14 @@ version = "0.7.1"
 Adapt = "3, 4"
 julia = "1.6"
 
+[deps]
+Adapt = "79e6a3ab-5dfb-504d-930d-738a2a938a0e"
```
Comment (wsmoses) [inline lib/EnzymeCore/Project.toml:11]: To confirm, this doesn't add the dep on 1.9+?

**Comment by wsmoses** [comment]
Okay. I'm a bit uneasy on this still, but I defer to your judgement.

**Review by wsmoses — APPROVED** [review]

File: lib/EnzymeCore/ext/AdaptExt.jl [hunk 3]
```diff
@@ -1,11 +1,11 @@
 module AdaptExt
-    using Adapt
-    using EnzymeCore
+isdefined(Base, :get_extension) ? (using Adapt) : (using ..Adapt)
```
Comment (devmotion) [inline lib/EnzymeCore/ext/AdaptExt.jl:2]: If the extension is loaded unconditionally on older Julia versions, you can just use
```suggestion
using Adapt
```
(and not load Adapt in EnzymeCore itself). See #954.

File: lib/EnzymeCore/ext/AdaptExt.jl [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
+isdefined(Base, :get_extension) ? (using EnzymeCore) : (using ..EnzymeCore)
```
Comment (devmotion) [inline lib/EnzymeCore/ext/AdaptExt.jl:3]:
```suggestion
using EnzymeCore
```

File: lib/EnzymeCore/src/EnzymeCore.jl [hunk 5]
```diff
@@ -257,4 +257,9 @@ function compiler_job_from_backend end
...
+    using Adapt
```
Comment (devmotion) [inline lib/EnzymeCore/src/EnzymeCore.jl:261]:
```suggestion
```

File: lib/EnzymeCore/Project.toml (hunk 2 above)
Comment (devmotion) [reply lib/EnzymeCore/Project.toml:11]: No, on Julia >= 1.9 it will still be only a weak dependency.

**Pull request merged**